lib/extension/dlisio/ext/io.hpp

```diff
@@ -26,64 +35,44 @@ struct record {
 
 class stream {
 public:
-    explicit stream( const std::string& path ) noexcept (false);
-
-    record  at( int i ) noexcept (false);
-    record& at( int i, record& ) noexcept (false);
-
-    void reindex( const std::vector< long long >&,
-                  const std::vector< int >& )
-        noexcept (false);
+    explicit stream( lfp_protocol* f) noexcept (false);
 
     void close();
+    int eof() const noexcept (true);
+
+    void seek( std::int64_t offset  ) noexcept (false);
```

Comment:
None of them? I don't understand what you are on about with this private stuff.  _All_ of them are used externally

Reply:
Oh right... You declared everything in the outside functions. 
OK. Do you expect caller from python to call read/seek/tell through core? 
If yes, I agree, leave them public.
If no, consider making them private and making extract and friends into actual "friend" functions. I think friend functions are allowed access to all the private members.

My main point in this "public" division is to figure out if it might or should be called from core.cpp.